src/index/hnsw/hnsw.cc

```diff
@@ -38,12 +38,8 @@ class HnswIndexNode : public IndexNode {
 
     Status
     Build(const DataSet& dataset, const Config& cfg) override {
-        auto res = this->Train(dataset, cfg);
-        if (res != Status::success) {
-            return res;
-        }
-        res = Add(dataset, cfg);
-        return res;
+        RETURN_IF_ERROR(Train(dataset, cfg));
```

Comment:
suggest to change like this:
        RETURN_IF_ERROR(Train(dataset, cfg));
        RETURN_IF_ERROR(Add(dataset, cfg));
        return Status::success;

Reply:
ditto